pkg/util/testing/client.go

```diff
@@ -64,3 +66,36 @@ func (b *builderIndexer) IndexField(ctx context.Context, obj client.Object, fiel
 func AsIndexer(builder *fake.ClientBuilder) client.FieldIndexer {
 	return &builderIndexer{ClientBuilder: builder}
 }
+
+type EventRecord struct {
+	Key       types.NamespacedName
+	EventType string
+	Reason    string
+	Message   string
+	// add annotations if ever needed
+}
+
+type EventTestRecorder struct {
```

Comment:
```suggestion
type EventRecorder struct {
```

we are already in the testing package.

Reply:
done